### PR TITLE
Closes #2357 Add better metatag defaults for Quickstart

### LINKED ISSUE #2357: Add better metatag defaults for quickstart.
## Motivation
Since the AZ SEO module was launched a lot of improvements to Quickstart have occurred without updating az_seo with new defaults that take advantage of those improvements.

### Is your feature request related to a problem? Please describe.
My site's metatags are incomplete when first installing quickstart.

## Proposed Resolution

Suggestions should be added as comments on this issue.

## Disclaimer

Since Quickstart already includes an incomplete default setting file for this config Arizona Digital owns the config until it is changed on an individual site, at which time, that site owns that configuration.

## More context

At the moment the defaults are managed in this [file](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_seo/config/quickstart/metatag.metatag_defaults.node.yml) and only contains three tags

```
  description: '[node:field_az_summary]'
  canonical_url: '[node:field_az_link:uri]'
  title: '[node:title] | [site:name]'
```

The Quckstart 1 implementation https://bitbucket.org/ua_drupal/ua_quickstart/src/7.x-1.x/modules/custom/uaqs_seo/

Historically, tokens were used in Quickstart 1 for defaults.  [Quickstart 2 also includes the token module](https://github.com/az-digital/az_quickstart/blob/main/composer.json#L108), and can be leveraged here.

## Current default permissions for metatags
Screenshot 2023-04-20 at 12 18 54 PM

### PR DESCRIPTION
**Pull request by bberndt-uaz**
## Description

Updates the default settings for the Metatag contrib module, primarily to:

- Enable social media previews
- Potentially improve SEO

Tasks to complete:

- [x] Figure out how to set both the canonical link and the og:url tag
   - Currently in Quickstart, if a node doesn't have an external link, the `[node:az-canonical-url]` token returns this URL string internally: `route:entity.node.canonical;node=17`. This string can't be used for a metatag value.
   - This PR introduces a new token to return an absolute URL instead, so that the `og:url` tag should always match the `rel="canonical"` link annotation. I tested this on a multidev of a multilingual site and confirmed that each translation of a page has a canonical link set for its specific URL (such as `/some-page` and `/es/some-page`).
   - More info on canonical links: https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls. In short, changing the behavior of what Quickstart sets as a canonical link should be a separate issue, which may alter the sitemap as well.
- [x] Confirm what image tags to use, depending on image size and style recommendations
   - The PR is currently using the _original_ thumbnail image (without cropping).
   - [Facebook recommendations](https://developers.facebook.com/docs/sharing/webmasters/images): at least 1200 x 630 px for best display on high res devices, 600 x 315 px to display link page posts with larger images. The minimum is 200 x 200 px and the size should be less than 8 MB. Image cropped to 1.91:1 aspect ratio.
   - [Twitter recommendations](https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/summary): minimum 144x144 px, maximum 4096x4096 px, less than 5 MB in size. Images cropped to square.
   - **Tentative decision:** Although we could potentially set the image style to the "Max 1300x1300" style, this would result in many image files being generated (unnecessarily in my opinion). So using the original image seems like the best option.
- [x] Confirm what changes to make to the robots tag, if any
   - The main change to make would be conditional: setting robots to noindex, nofollow for nodes with an external link. This can be a follow-up PR.
- [x] Fix duplicate Generator tags
   - This is an [issue with the Metatag module](https://www.drupal.org/project/metatag/issues/2735195#comment-14531696): the Generator tag from Drupal does not get overridden. A Generator tag for "Arizona Quickstart" will not be included in this PR.

## Related issues

#2357

## How to test

Probo review site: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-3255.probo.build

1. Log into the Probo site and update pages, news, etc. with summary text and other content for testing. ([Example](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-3255.probo.build/news/could-mans-best-friend-be-mans-best-medicine))
2. View the source of the page and check the <head> element to confirm that the `og` tags are present and that the `og:url` tag matches the `rel="canonical"` link annotation.
3. Use the tools below to test the social media previews for various URLs:

- https://metatags.io
- [Facebook Sharing Debugger](https://developers.facebook.com/tools/debug/)
- [Twitter Card Validator](https://cards-dev.twitter.com/validator)

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — DISMISSED** [review]

File: modules/custom/az_core/az_core.module [hunk 1]
```diff
@@ -259,14 +259,14 @@ function az_core_canonical_summary_link_title(array $data) {
...
- *   The node's canononical URI.
+ *   The node's canonical URI.
  */
 function az_core_canonical_url(array $data) {
   /** @var \Drupal\node\NodeInterface $node */
   $node = $data['node'];
   /** @var \Drupal\Core\Url $url */
   $url = $node->toUrl();
-  $uri = $url->toUriString();
+  $uri = $url->setOption('absolute', TRUE)->toString();
```
Comment (trackleft) [inline modules/custom/az_core/az_core.module:269]: I think I. am ok with changing this.

**Comment by bberndt-uaz** [comment]
I can add a `[node:az-canonical-url:absolute]` token instead of changing the existing one.

**Comment by bberndt-uaz** [comment]
Adding a new `[node:az-canonical-url:absolute]` chained token would require defining a new token type and making `absolute` a new token of that type. From what I could tell, this further chaining could have a performance impact and it generally seems unnecessary for a single additional token, so I decided to simply add a new `[node:az-canonical-absolute-url]` token that also uses the `node` type.

File: modules/custom/az_core/az_core.module [hunk 2]
```diff
@@ -250,23 +258,31 @@ function az_core_canonical_summary_link_title(array $data) {
...
   /** @var \Drupal\Core\Url $url */
   $url = $node->toUrl();
-  $uri = $url->toUriString();
+
+  if ($absolute) {
+    $uri = $url->setOption('absolute', TRUE)->toString();
+  }
+  else {
+    $uri = $url->toUriString();
+  }
```
Comment (bberndt-uaz) [inline modules/custom/az_core/az_core.module:285]:
```suggestion
  $uri = $absolute ? $url->setOption('absolute', TRUE)->toString() : $url->toUriString();
```

**Review by trackleft — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**